Add hostedDomain config

GuilhermeBarile opened the pull request:
Add hostedDomain (hd) option for the gcloud provider. Also pull from upstream as well.

jondoveston committed:
Add a google group validation

In the same way that the email domain and the email whitelist is checked, the
email is now checked for google group membership. To check an email is a member
of a google group requires access to the gcloud API for the whole google domain.
The official Google Gcloud API client library is used and requires extra
configuration:

- GoogleDomain is the domain managed by GCloud
- GoogleGroup is a comma delimited list of the required group emails
- GoogleApplicationCredentials is path to service account JSON credentials file
- GoogleActingAdminEmail is the email an admin account with domain access
- GoogleExpirySeconds is the number of seconds the API response is cached

The service account requires domain-wide delegation of authority as discussed
here https://developers.google.com/admin-sdk/directory/v1/guides/delegation.
It also requires the oauth scope
https://www.googleapis.com/auth/admin.directory.group.readonly to be able to
read the groups of a member email. This service account has to impersonate an
admin account to access the admin APIs.

The default google groups required for authorization is specified in the
GoogleGroup variable, but can also be specified in a Rule like:

--rule.1.action=allow
--rule.1.rule=Host(`example.com`)
--rule.1.google_groups=team@example.com

The order of the rules is significant and controls the priority of the rule
matching.

thomseddon committed:
Remove double brackets typo in readme (#218)

thomseddon committed:
Switch to Github Actions for CI (#219)

hesstobi committed:
Make listen port configurable (#230)

Co-authored-by: Tobias Hess <tobias.hess@energiekoppler.com>

Commit:
Add GitHub Actions workflow for creating binaries for releases (#184) (#199)

* Add GitHub Actions workflow for creating binaries for releases
* Add sentence about binary files to README
* Cleanup + nicer way querying GitHub API

thomseddon committed:
Allow to be run without middleware + improve request reading consistency (#217)

Prior to this change, the request URI was only ever read from the
X-Forwarded-Uri header which was only set when the container was
accessed via the forwardauth middleware. As such, it was necessary
to apply the treafik-forward-auth middleware to the treafik-forward-auth
container when running auth host mode.
This is a quirk, unnecessary complexity and is a frequent source of
configuration issues.

GuilhermeBarile commented:
@jondoveston should we parameterize this?

jondoveston reviewed (changes requested):
Yes I think this would be better as a parameter.

jondoveston committed:
Add the HostedDomain option to the gcloud provider

The HostedDomain string is passed to gcloud oauth as an URI parameter and will
only show accounts from that domain.

https://developers.google.com/identity/protocols/oauth2/openid-connect#hd-param

jondoveston committed:
Merge branch 'toptal' into hd-toptal.com

GuilhermeBarile commented:
:+1:
